src/trace/dma-trace.c

```diff
@@ -269,6 +269,13 @@ static int dma_trace_start(struct dma_trace_data *d)
 	uint32_t elem_size, elem_addr, elem_num;
 	int err = 0;
 
+	/* DMA Controller initialization is platform-specific */
+	if (!d || !d->dc.dmac) {
+		mtrace_printf(LOG_LEVEL_ERROR,
+			      "dma_trace_start failed: no DMAC!");
+		return 1;
```

Comment:
any particular reason for returning 1 and not 0?

Reply:
Because it's an error never supposed to happen?

Reply:
sorry not following. EVen if its an error thats not supposed to happen, why not return -ENODEV?

Reply:
My bad it should be negative indeed.

(It shouldn't be 0 as you suggested earlier, that confused me)